api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+    return average / ranks.length;
+  } else {
+    const gameRanks = await getGameRanks(ranks[0].game_id);
+    ranks.forEach((rank) => {
+      average += indexOf(gameRanks, rank.text_rank);
+    });
+    return gameRanks[average / ranks.length];
+  }
+};
+
+/**
+ * @swagger
+ * /ranks:
+ *   get:
+ *     summary: Get all ranks
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks of all users from the database.
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+router.get("/", async (req: Request, res: Response) => {
+  try {
+    const result = await pool.query("SELECT * FROM user_ranks");
+    res.json(result.rows);
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/user:
+ *   get:
+ *     summary: Get all ranks of a user on a game
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks for a specific user on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+router.get("/user", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId } = req.query;
+
+    const result = await pool.query(
+      "SELECT * FROM user_ranks WHERE user_id = $1 AND game_id = $2",
+      [userId, gameId]
+    );
+    res.json(result.rows);
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/user/{userId}:
+ *   get:
+ *     summary: Get rank by user id
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks for a specific user from the database.
+ *     parameters:
+ *       - in: path
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: page
+ *         description: Page number for pagination
+ *         schema:
+ *           type: integer
+ *         default: 1
+ *       - in: query
+ *         name: pageSize
+ *         description: Number of logs per page
+ *         schema:
+ *           type: integer
+ *         default: 10
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+router.get("/user/:userId", async (req: Request, res: Response) => {
```

Comment:
After some googling, I do believe this should be `GET /users/{user_id}/ranks` not `GET /ranks/user/{user_id}/`, since this follow the hierarchical structure of the DB. Like a user has ranks, rank doesn't have users. Does this make sense?